app/assets/javascripts/sufia/save_work/save_work_control.es6

```diff
@@ -95,6 +96,13 @@ export class SaveWorkControl {
   validateFiles() {
     if (!this.isNew || this.uploads.hasFiles) {
       this.requiredFiles.check()
+
+      if (this.uploads.hasFiles && this.mustAgreeAgain) {
```

Comment:
I don't think this code belongs in this method because the code you added is not validating that they have files on the page.